### PR TITLE
Fix h264 payloader stream format generation

### PR DESCRIPTION
**Pull request by Qizot**
It happened that when only PPS parameters changed we triggered new stream format generation. 

In this case, SPS values may have not been changed therefore they are absent in the currently processed buffer. This eventually results in an invalid call to `hd(sps)` where `sps` value is empty so the payloader raises.

### CODE REVIEW

File: lib/membrane_mp4/payloader/h264.ex [hunk 1]
```diff
@@ -128,4 +134,7 @@ defmodule Membrane.MP4.Payloader.H264 do
   defp encode_parameter_sets(pss) do
     Enum.map_join(pss, &<<byte_size(&1)::16-integer, &1::binary>>)
   end
+
+  defp fetch_parameters_set([] = _new_ps, ps), do: ps
+  defp fetch_parameters_set(ps, _old_ps), do: ps
```
Comment (mat-hek) [inline lib/membrane_mp4/payloader/h264.ex:139]:
```suggestion
  defp fetch_parameters_sets([] = _new_ps, ps), do: ps
  defp fetch_parameters_sets(ps, _old_ps), do: ps
```

**Review by mat-hek — APPROVED** [review]

File: lib/membrane_mp4/payloader/h264.ex (hunk 1 above)
Comment (Qizot) [reply lib/membrane_mp4/payloader/h264.ex:139]: It is actually a single parameter set 😛 (either sps or pps).

**Review by varsill — COMMENTED** [review]
I am afraid that would still fail in case of such a stream: ` [pps] ... [pps] ` (the `state.pps` would be equal to `nil` and we would effectively try to do `hd(state.pps`)

**Review by varsill — APPROVED** [review]

File: lib/membrane_mp4/payloader/h264.ex (hunk 1 above)
Comment (mat-hek) [reply lib/membrane_mp4/payloader/h264.ex:139]: I think it doesn't have to be, we retrieve a list [here](https://github.com/membraneframework/membrane_mp4_plugin/pull/76/files#diff-6106dedc0d1aad514f4602eedffb94b1299781db253a77ab1ebe4eb2b2b6ba46R54), also there can be multiple sps and multiple pps in AVCC, also there's [encode_parameter_sets](https://github.com/membraneframework/membrane_mp4_plugin/pull/76/files#diff-6106dedc0d1aad514f4602eedffb94b1299781db253a77ab1ebe4eb2b2b6ba46R134)

**Pull request merged**